dev/com.ibm.ws.cdi.ejb.common/src/com/ibm/ws/cdi/ejb/impl/EjbDescriptorImpl.java

```diff
@@ -225,7 +206,18 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        return "EjbDescriptor: " + this.ejbJ2EENameString;
+        return PREFIX + this.ejbJ2EENameString;
     }
 
+    public static String parseJ2EEName(EjbDescriptor<?> desc) {
+        String j2eeName = null;
+        String ejbDescString = desc.toString();
+        if (ejbDescString.startsWith(PREFIX)) {
+            j2eeName = ejbDescString.substring(PREFIX.length());
```

Comment:
I have not tested. Should this be PREFIX.length() -1 ?

Reply:
I have not tested either ... but no. If PREFIX was length 0 then you would want to start at index 0. If PREFIX was length 1 then you would want to start at index 1.

Reply:
ok. makes sense